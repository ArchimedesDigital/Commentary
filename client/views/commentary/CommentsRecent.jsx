import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

CommentsRecent = React.createClass({

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(baseTheme) };
	},
	render() {
		return (
			<section className="background-gray recent-comments">

				<div className="container">
					<div className="row">
						<div className="col-sm-8 col-sm-offset-2 text-center">
							<h3 className=" uppercase">
								Recently from the Commentary
							</h3>
							<i className="mdi mdi-format-quote quote-icon" />
							<div className="text-slider slider-arrow-controls">
								<ul className="slides">
									<li>
										<p >
											Incidentally, it is also clear just from our Iliad that there could be
											another starting point for the tale of Achilles' wrath. When a bit later in
											Iliad 1 Achilles answers his mother's request to ἐξαυδᾶν 'speak out' what
											has befallen him, he begins . . .
										</p>
										<h4>&mdash; Douglas Frame, Iliad 1.4-6</h4>

									</li>
								</ul>
							</div>
						</div>
					</div>
				</div>

			</section>


		);
	},
});
