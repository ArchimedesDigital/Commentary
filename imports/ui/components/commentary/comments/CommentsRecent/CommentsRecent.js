import React from 'react';
import { Meteor } from 'meteor/meteor';
import { Session } from 'meteor/session';
import { createContainer } from 'meteor/react-meteor-data';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Slider from 'react-slick';

// api
import Comments from '/imports/models/comments';

// lib:
import muiTheme from '/imports/lib/muiTheme';
import Utils from '/imports/lib/utils';


/*
	helpers
*/
const settings = {
	dots: true,
	arrows: false,
	infinite: true,
	autoplay: true,
	speed: 300,
	autoplaySpeed: 10000,
	slidesToShow: 1,
	slidesToScroll: 1,
};

/*
	BEGIN CommentsRecent
*/
const CommentsRecent = ({ comments }) => {
	if (!comments) {
		return null;
	}
	return (
		<section className="background-gray recent-comments">
			<div className="container">
				<div className="row">
					<div className="col-sm-8 col-sm-offset-2 text-center">
						<h3 className=" uppercase">
							Recently from the Commentary
						</h3>
						<i className="mdi mdi-format-quote quote-icon" />
						<Slider {...settings}>
							{comments.map(comment => (
								<div
									key={comment._id}
								>
									<p
										dangerouslySetInnerHTML={{
											__html: Utils.trunc(comment.revisions[comment.revisions.length - 1].text, 300),
										}}
									/>
									<h4>
										-
										{comment.commenters.map(commenter => (
											` ${commenter.name},`
										))}
										{` ${comment.work.title} ${
										comment.subwork.title}.${comment.lineFrom}-${
										comment.lineFrom + comment.nLines}`}
									</h4>
								</div>
							))}
						</Slider>
					</div>
				</div>
			</div>
		</section>
	);
};
CommentsRecent.propTypes = {
	comments: React.PropTypes.arrayOf(React.PropTypes.shape({
		_id: React.PropTypes.string.isRequired,
		revisions: React.PropTypes.arrayOf(React.PropTypes.shape({
			text: React.PropTypes.string.isRequired,
		})).isRequired,
		commenters: React.PropTypes.arrayOf(React.PropTypes.shape({
			name: React.PropTypes.string.isRequired,
		})).isRequired,
		work: React.PropTypes.shape({
			title: React.PropTypes.string.isRequired,
		}).isRequired,
		subwork: React.PropTypes.shape({
			title: React.PropTypes.string.isRequired,
		}).isRequired,
		lineFrom: React.PropTypes.number.isRequired,
		nLines: React.PropTypes.number.isRequired,
	})),
};
CommentsRecent.defaultProps = {
	comments: null,
};
/*
	END CommentsRecent
*/

export default createContainer(() => {
	const handle = Meteor.subscribe('comments.recent.tenant', Session.get('tenantId'), 3);
	let comments = [];
	if (handle.ready()) {
		comments = Comments.find({}, {
			sort: {
				updated: -1,
			},
		}).fetch();
	}
	return {
		comments,
	};
}, CommentsRecent);
