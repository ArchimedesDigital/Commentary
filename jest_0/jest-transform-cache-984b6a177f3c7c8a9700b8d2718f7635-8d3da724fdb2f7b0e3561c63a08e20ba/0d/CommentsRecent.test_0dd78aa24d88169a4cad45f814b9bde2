/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_CommentsRecent = require('./CommentsRecent'); /* istanbul ignore next */var _CommentsRecent2 = _interopRequireDefault(_CommentsRecent);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('CommentsRecent', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_CommentsRecent2.default, /* istanbul ignore next */null)).
		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: