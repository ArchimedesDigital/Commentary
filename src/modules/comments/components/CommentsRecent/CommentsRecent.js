import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { compose } from 'react-apollo';
import Slider from 'react-slick';


// lib:
import commentsQuery from '../../graphql/queries/comments';
import Utils from '../../../../lib/utils';


/*
	helpers
*/
const settings = {
	dots: true,
	arrows: false,
	infinite: true,
	autoplay: true,
	speed: 300,
	autoplaySpeed: 10000,
	slidesToShow: 1,
	slidesToScroll: 1,
};

/*
	BEGIN CommentsRecent
*/
class CommentsRecent extends Component {
	constructor(props) {
		super(props);
		this.state = {};
	}
	componentWillReceiveProps(nextProps) {
		this.setState({
			comments: nextProps.commentsQuery.loading ? [] : nextProps.commentsQuery.comments.slice(0,3)
		});
	}
	render() {
		const { comments } = this.state;
		if (!comments) {
			return null;
		}
		return (
			<section className="background-gray recent-comments">
				<div className="container">
					<div className="row">
						<div className="col-sm-8 col-sm-offset-2 text-center">
							<h3 className=" uppercase">
								Recently from the Commentary
							</h3>
							<i className="mdi mdi-format-quote quote-icon" />
							<Slider {...settings}>
								{comments.map(comment => (
									<div
										key={comment._id}
									>
										<p
											dangerouslySetInnerHTML={{
												__html: Utils.trunc(comment.revisions[comment.revisions.length - 1].text, 300),
											}}
										/>
										<h4>
											-
											{comment.commenters.map(commenter => (
												` ${commenter.name},`
											))}
											{comment.work ? ` ${comment.work.title} ${
											comment.subwork.title}.${comment.lineFrom}-${
											comment.lineFrom + comment.nLines}`: ''}
										</h4>
									</div>
								))}
							</Slider>
						</div>
					</div>
				</div>
			</section>
		);
	}
}
CommentsRecent.propTypes = {
	commentsQuery: PropTypes.object
};
CommentsRecent.defaultProps = {
	comments: null,
};

export default compose(commentsQuery)(CommentsRecent);
