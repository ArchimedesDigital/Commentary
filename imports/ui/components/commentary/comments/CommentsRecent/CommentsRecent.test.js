import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommentsRecent from './CommentsRecent';

describe('CommentsRecent', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<CommentsRecent />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
