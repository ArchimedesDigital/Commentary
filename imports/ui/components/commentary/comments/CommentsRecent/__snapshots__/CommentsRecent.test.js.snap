// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsRecent renders correctly 1`] = `null`;
